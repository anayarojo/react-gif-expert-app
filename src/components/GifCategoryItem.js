import React from 'react';
import PropTypes from 'prop-types';


export const GifCategoryItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__fadeIn animate__slower">
            <img src={ url } alt={title} />
            <p className="mb-5">{ title }</p>
        </div>
    )
}

GifCategoryItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

export default GifCategoryItem;
