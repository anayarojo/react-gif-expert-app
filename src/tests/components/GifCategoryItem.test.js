import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';

import GifCategoryItem from '../../components/GifCategoryItem';

describe('Test for <GifCategoryItem />', () => {
    
    const title = 'Title';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifCategoryItem title={ title } url={ url } />);

    test('Should show <GifCategoryItem /> successfully', () => {
    
        expect(wrapper).toMatchSnapshot();
    });

    test('Should has a paragraph with title prop', () => {
    
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Should has an image equal to url prop', () => {
    
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
    });

    test('Should has an image with alt equal to title prop', () => {
    
        const img = wrapper.find('img');
        expect(img.prop('alt')).toBe(title);
    });

    test('Should has "animate__fadeIn" class', () => {
    
        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);
    });

});
