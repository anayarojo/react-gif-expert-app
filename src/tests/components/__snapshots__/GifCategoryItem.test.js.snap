// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifCategoryItem /> Should show <GifCategoryItem /> successfully 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__slower"
>
  <img
    alt="Title"
    src="https://localhost/something.jpg"
  />
  <p
    className="mb-5"
  >
    Title
  </p>
</div>
`;
